### PR TITLE
Agent: Refactor get_routes to resolve C901

### PR DESCRIPTION
**Pull request by ilija-lazoroski**
# What does this PR do?

Fixes part of #1076 .

Add any further explanations here.

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [x] Was the CHANGELOG.md updated to reflect the changes?
* [x] Was the documentation framework updated to reflect the changes?
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [x] Added relevant unit tests?
* [x] Do all unit tests pass?
* [ ] Do all end-to-end tests pass?
* [x] Any other testing performed?
    > Tested by {Running the Monkey locally with relevant config/running Island/...}
* [ ] If applicable, add screenshots or log transcripts of the feature working

### CODE REVIEW

**Review by shreyamalviya — COMMENTED** [review]
Did you test this? Fill out the checklist.

**Review by shreyamalviya — APPROVED** [review]
Approving as long as this is manually tested

File: monkey/infection_monkey/network/info.py [hunk 1]
```diff
@@ -35,60 +38,72 @@ class NetworkAddress:
     domain: Optional[str]
 
 
-if is_windows_os():
-
-    def get_routes():
+def get_routes() -> List[Tuple[int, int, str, bytes, str]]:
```
Comment (cakekoa) [inline monkey/infection_monkey/network/info.py:41]: Can we add a docstring? Also, it would be nice to have a NamedTuple or dataclass instead of a tuple

Comment (mssalvatore) [reply monkey/infection_monkey/network/info.py:41]: +1

File: monkey/infection_monkey/network/info.py [hunk 2]
```diff
@@ -35,60 +38,72 @@ class NetworkAddress:
...
+
+
+def _extract_network_info(line: str) -> Tuple[bytes, bytes, bytes, int, bytes]:
+    values = [var.encode() for var in line.split()]
+    iff: bytes = values[0]
+    dst: bytes = values[1]
+    gw: bytes = values[2]
+    flags: int = int(values[3], 16)
+    msk: bytes = values[7]
+    return iff, dst, gw, flags, msk
```
Comment (mssalvatore) [inline monkey/infection_monkey/network/info.py:95]: Another good candidate for a dataclass. But we don't have to go overboard with this refactoring. Our primary goal is cyclomatic complexity, which looks to be resolved.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**